### PR TITLE
AWS- EC2 - add aws api module

### PR DESCRIPTION
**Pull request by tallieber**
## Status
- [x] In Progress
- [ ] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/34707

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 1]
```diff
@@ -2979,241 +2931,256 @@ def allocate_hosts_command(args):
...
+        aws_secret_access_key = params.get('secret_key')
+        verify_certificate = not params.get('insecure', True)
+        timeout = params.get('timeout')
+        retries = params.get('retries') or 5
+        validate_params(aws_default_region, aws_role_arn, aws_role_session_name, aws_access_key_id,
+                        aws_secret_access_key)
+        aws_client = AWSClient(aws_default_region, aws_role_arn, aws_role_session_name, aws_role_session_duration,
+                               aws_role_policy, aws_access_key_id, aws_secret_access_key, verify_certificate, timeout,
+                               retries)
+
```
Comment (bakatzir) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:2966]: please call also demiato.command() and demisto.args() once here.

File: Packs/AWS-EC2/ReleaseNotes/1_2_0.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### AWS - EC2
+- Internal code improvements.
```
Comment (bakatzir) [inline Packs/AWS-EC2/ReleaseNotes/1_2_0.md:4]: can rm. not needed when we have detailed rn. any pr contains also some internal code improvements..

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 3]
```diff
@@ -2979,241 +2931,256 @@ def allocate_hosts_command(args):
...
 
         elif demisto.command() == 'aws-ec2-allocate-hosts':
-            allocate_hosts_command(demisto.args())
+            allocate_hosts_command(demisto.args(), aws_client)
 
         elif demisto.command() == 'aws-ec2-release-hosts':
-            release_hosts_command(demisto.args())
+            release_hosts_command(demisto.args(), aws_client)
 
     except ResponseParserError as e:
```
Comment (bakatzir) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:3185]: this also can be rmed, handled in the apimodule

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml [hunk 4]
```diff
@@ -4801,6 +4817,17 @@ script:
...
+      default: false
+      description: Indicates that the fleet launches all Spot Instances into a single
+        Availability Zone.
+      isArray: false
+      name: SpotSingleInstanceType
+      predefined:
+      - 'True'
+      - 'False'
+      required: false
+      secret: false
```
[the remark this answers is not in the document]
Comment (tallieber) [reply Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml:4830]: This is a mistake caused while I merged from master

**Review by bakatzir — APPROVED** [review]

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml [hunk 5]
```diff
@@ -46,6 +46,23 @@ configuration:
...
+    after a comma (for example 60,10). If a connect timeout is not specified, a default
+    of 10 second will be used.
+  defaultvalue: 60,10
+  required: false
+  type: 0
+- display: Retries
+  name: retries
+  defaultvalue: 5
+  additionalinfo: The maximum retry attempts when connection or throttling errors
+    are encountered. Set to 0 to disable retries. The default value is 5 and the limit is for 10 retries.
```
Comment (bakatzir) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml:62]:
```suggestion
    are encountered. Set to 0 to disable retries. The default value is 5 and the limit is 10 retries.
```

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml [hunk 6]
```diff
@@ -46,6 +46,23 @@ configuration:
...
+    Please considered when you increase the number of retries, execution time will increased respectively.
```
Comment (bakatzir) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml:63]:
```suggestion
  additionalinfo: The maximum retry attempts when connection or throttling errors
    are encountered. Set to 0 to disable retries. The default value is 5 and the limit is 10.
    When increasing the number of retries, execution time will increased respectively.
```

Comment (tallieber) [reply Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.yml]: @ShirleyDenkberg reword it, now the final description should apperar

**Pull request merged**